### DISCUSSION TITLE
my permissions and role deleted automatically

### DISCUSSION THREAD

**Discussion by moecods** [spatie/laravel-permission#1858] [Q&A] [answered]
hi, when i save role and permission in database table after 1h They are all deleted automatically!
How to avoid delete permissions and roles

**Comment by erikn69** [comment]
Impossible, they cannot be automatically deleted from the database, I have 358 permissions and 15 roles and none have ever been deleted, it must be from your implementation

**Comment by moecods** [comment, marked as answer]
i found my problem, i define permission:init console command and in __construct drop permission and role database table!
and when i run any command cosole it is run!
